app/controllers/trainees/training_routes_controller.rb

```diff
@@ -8,12 +8,22 @@ class TrainingRoutesController < ApplicationController
     def edit; end
 
     def update
-      trainee.update!(trainee_params)
-      redirect_to page_tracker.last_origin_page_path || trainee_path(trainee)
+      return redirect_url if route_not_changed?
+
+      ActiveRecord::Base.transaction do
```

Comment:
I'd push this further down into either:

- A service which will update the trainee params and clear the attributes
- Into the model via a custom method to handle resetting other attributes based on route changes

The model might work fine for now but i can see this growing for more cases in the future, so a service might better long term.

Then your unit test can just deal with the logic of that class/method.

Reply:
All done 👍